include/simdjson/generic/ondemand/value_iterator-inl.h

```diff
@@ -86,7 +85,7 @@ simdjson_warn_unused simdjson_really_inline simdjson_result<bool> value_iterator
     // this object iterator will blithely scan that object for fields.
     if (_json_iter->depth() < depth() - 1) { return OUT_OF_ORDER_ITERATION; }
 #endif
-    has_value = false;
+    return false;
```

Comment:
This is probably performance neutral (and clearer), but just checking: this change doesn't have an actual effect, right? `while (has_value)` should do nothing, and then we return false at the end?

Reply:
The next bit is `while(has_value){` which, indeed, does not get executed if set `has_value = false` but that loop end mid-course in the function and then after that you have the rest of the function where you reenter the object and we try to restart the iteration... that part gets executed even if `has_value = false` is set.